### ISSUE TITLE
CA2208 false positive

### ISSUE THREAD

**Issue by Abrynos** [dotnet/roslyn#70824]
## **Version Used**

4.8.0-3.23524.11 (f43cd10b) (installed with official dotnet release 8.0.100

## **Steps to Reproduce**:

1. Install dotnet 8.0.100 (not reproducible with net7.0
2. Create a new app: `dotnet new console`
3. Put following inside the newly generated `.csproj` file:
```xml
<PropertyGroup>
    <Nullable>enable</Nullable>
    <AnalysisMode>AllEnabledByDefault</AnalysisMode>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
</PropertyGroup>
```
4. Replace all of `Program.cs` with following content:
```csharp
using System;

namespace Test;

internal sealed class SomeClass
{
    internal object? SomeProperty { get; init; }
}

public static class Program
{
    private static void Test(SomeClass someParameter)
    {
        ArgumentNullException.ThrowIfNull(someParameter);

        if (someParameter.SomeProperty is not int i)
        {
            throw new ArgumentException($"{nameof(someParameter)}.{nameof(someParameter.SomeProperty)} is not {nameof(Int32)}", nameof(someParameter));
        }

        Console.WriteLine($"test {i}");
    }

    public static void Main(string[] args) => Test(new SomeClass() { SomeProperty = 42 });
}

```
5. Compile using `dotnet build`

## **Diagnostic Id**

CA2208

## **Expected Behavior**

The build completes successfully because the parameter name is correctly put into the constructor of `ArgumentException`

## **Actual Behavior**

The build fails due to `CA2208`

**Comment by Abrynos** [comment]
Seems like I posted in the wrong repository. I'm sorry. This is related to dotnet/roslyn-analyzers#6863
